packages/core/src/modules/dids/methods/sov/IndySdkSovDidRegistrar.ts

```diff
@@ -170,6 +167,71 @@ export class SovDidRegistrar implements DidRegistrar {
       },
     }
   }
+
+  public async registerPublicDid(
+    agentContext: AgentContext,
+    submitterDid: string,
+    targetDid: string,
+    verkey: string,
+    alias: string,
+    role?: Indy.NymRole
+  ) {
+    const pool = this.indyPoolService.ledgerWritePool
+
+    try {
+      this.logger.debug(`Register public did '${targetDid}' on ledger '${pool.id}'`)
+
+      const request = await this.indy.buildNymRequest(submitterDid, targetDid, verkey, alias, role || null)
+
+      const response = await this.indyPoolService.submitWriteRequest(agentContext, pool, request, submitterDid)
+
+      this.logger.debug(`Registered public did '${targetDid}' on ledger '${pool.id}'`, {
+        response,
+      })
+
+      return targetDid
+    } catch (error) {
+      this.logger.error(`Error registering public did '${targetDid}' on ledger '${pool.id}'`, {
+        error,
+        submitterDid,
+        targetDid,
+        verkey,
+        alias,
+        role,
+        pool: pool.id,
+      })
+
+      throw error
+    }
+  }
+
+  public async setEndpointsForDid(
+    agentContext: AgentContext,
+    did: string,
+    endpoints: IndyEndpointAttrib
+  ): Promise<void> {
+    const pool = this.indyPoolService.ledgerWritePool
```

Comment:
Same here, should use pool based on the indyNamespace variable. I think for both methods we can pass the pool as an argument to this method.